src/main.rs

```diff
@@ -1,3 +1,186 @@
-fn main() {
-    println!("Hello, world!");
+use actix_web::{web, App, HttpServer, Result};
+use serde::Deserialize;
+extern crate serde_json;
+use serde_json::Value;
+
+//extern crate reqwest;
+use reqwest;
+// This struct represents state
+
+#[derive(Deserialize)]
+struct ContractCall {
+    contract: String,
+    txn: String,
+}
+
+async fn get_txn_info(txn: &String) -> String {
+    let res = reqwest::get(format!(
+        "https://blockscout.com/eth/mainnet/api?module=transaction&action=gettxinfo&txhash={}",
+        txn
+    ))
+    .await;
+
+    let res = match res {
+        Ok(r) => r,
+        Err(err) => panic!("{}", err),
+    };
+
+    let res = res.text().await;
+
+    let res = match res {
+        Ok(r) => r,
+        Err(err) => panic!("{}", err),
+    };
+
+    res
+}
+
+fn get_txn_input_bytes(txn: &String) -> String {
+    let data: Value = serde_json::from_str(&txn).unwrap();
```

Comment:
`?`

Reply:
basically you should replace all panics and unwraps to propagating the errors with `?`

Reply:
https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html